CMakeLists.txt

```diff
@@ -243,31 +265,41 @@ endif()
 option(USE_OPTIMIZED_BLAS "Whether or not to use an optimized BLAS library instead of included netlib BLAS" OFF)
 
 # Check the usage of the user provided BLAS libraries
-if(BLAS_LIBRARIES)
-  include(CheckFortranFunctionExists)
-  set(CMAKE_REQUIRED_LIBRARIES ${BLAS_LIBRARIES})
-  CHECK_FORTRAN_FUNCTION_EXISTS("dgemm" BLAS_FOUND)
-  unset(CMAKE_REQUIRED_LIBRARIES)
-  if(BLAS_FOUND)
-    message(STATUS "--> BLAS supplied by user is WORKING, will use ${BLAS_LIBRARIES}.")
+if(USE_OPTIMIZED_BLAS)
+  if(BLAS_LIBRARIES)
+    include(CheckFortranFunctionExists)
+    set(CMAKE_REQUIRED_LIBRARIES ${BLAS_LIBRARIES})
+    CHECK_FORTRAN_FUNCTION_EXISTS("dgemm" BLAS_FOUND)
+    unset(CMAKE_REQUIRED_LIBRARIES)
+    if(BLAS_FOUND)
+      message(STATUS "--> BLAS supplied by user is WORKING, will use ${BLAS_LIBRARIES}.")
+    else()
+      message(ERROR "--> BLAS supplied by user is not WORKING, CANNOT USE ${BLAS_LIBRARIES}.")
+      message(ERROR "-->     Fix BLAS_LIBRARIES or ensure that USE_OPTIMIZED_BLAS is off for REFERENCE BLAS.")
+    endif()
+  # User did not provide a BLAS Library but specified to search for one
   else()
-    message(ERROR "--> BLAS supplied by user is not WORKING, CANNOT USE ${BLAS_LIBRARIES}.")
-    message(ERROR "-->     Will use REFERENCE BLAS (by default)")
-    message(ERROR "-->     Or Correct your BLAS_LIBRARIES entry ")
-    message(ERROR "-->     Or Consider checking USE_OPTIMIZED_BLAS")
+    find_package(BLAS)
+  endif()
+else()
+  if(BLAS_LIBRARIES)
+    message(FATAL_ERROR "USE_OPTIMIZED_BLAS unset but BLAS provided. Accident?")
   endif()
-
-# User did not provide a BLAS Library but specified to search for one
-elseif(USE_OPTIMIZED_BLAS)
-  find_package(BLAS)
 endif()
 
 # Neither user specified or optimized BLAS libraries can be used
 if(NOT BLAS_FOUND)
+  if(USE_OPTIMIZED_BLAS)
+    message(FATAL_ERROR "USE_OPTIMIZED_BLAS set but no usable BLAS provided.")
+  endif()
   message(STATUS "Using supplied NETLIB BLAS implementation")
   add_subdirectory(BLAS)
   set(BLAS_LIBRARIES ${BLASLIB})
+  set(BLAS_PCLIBS -l${BLASLIB})
```

Comment:
I don't agree with this.

Reply:
Care to elaborate? Is this the same point about `Libs` and `Requires` above?

Reply:
It is the same point as above. Don't add libs of other pc files into an unrelated pc file. Rather directly depend on the pc file in `Requires`.